python/ray/serve/_private/router.py

```diff
@@ -799,6 +896,8 @@ def maybe_start_scheduling_tasks(self):
             self._scheduling_tasks.add(
                 self._loop.create_task(self.fulfill_pending_requests())
             )
+        if tasks_to_start > 0:
+            self.num_scheduling_tasks_gauge.set(len(self._scheduling_tasks))
```

Comment:
why do we set same metric into two places?

Reply:
I set the metric whenever a task is added or removed.